### DISCUSSION TITLE
Including Fpp Models in XML Topology

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#1477] [Q&A]
In the Fprime v3.0 Migration Guide, it says 

"There is one caveat that users should be aware of: topology models. If a project wishes to use FPP as a modeling language for a given topology, they must ensure that all components used by that topology are also modeled with FPP. XML topologies can include both components modeled in XML and FPP so that projects can begin to migrate to FPP models without needing to convert everything."

However, I am unsure of how to actually include fpp models into an XML topology. Are there any examples available?

**Comment by Joshua-Anderson** [comment]
FPP components are compiled into XML components. I forget the exact naming convention xml files but you should be able to include the xml equivalent of your FPP component in your XML topology.
